dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -690,6 +690,22 @@ private static (Func<KernelFunction, Kernel, KernelArguments, CancellationToken,
 
         var converter = GetConverter(type);
 
+        var jsonStringParsers = new Dictionary<Type, Func<string, object>>(12)
```

Comment:
Consider using a static field instead, so the dictionary is not created each time the method is called.